packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,465 @@
+    uint256 vestingMaxPausePeriod,
+    address registryAddress
+  ) public {
+    uint256 numPeriods = vestingAmount.div(vestAmountPerPeriod);
+    require(numPeriods >= 1, "There must be at least one vesting period");
+    require(
+      numPeriods.mul(vestAmountPerPeriod) == vestingAmount,
+      "Vesting amount per period and total vesting amount are inconsistent"
+    );
+    require(vestingBeneficiary != address(0), "Beneficiary is the zero address");
+    require(vestingAmount > 0, "Vesting amount must be positive");
+    require(vestingCliff <= vestingPeriodSec, "Vesting cliff is longer than vesting duration");
+    require(
+      vestingStartTime.add(numPeriods.mul(vestingPeriodSec)) > block.timestamp,
+      "Vesting end time must be in the future"
+    );
+    require(vestingMaxPausePeriod > 0, "maximum pause period must be greater than zero");
+
+    setRegistry(registryAddress);
+
+    vestingSchedule.vestingAmount = vestingAmount;
+    vestingSchedule.vestAmountPerPeriod = vestAmountPerPeriod;
+    vestingSchedule.vestingPeriodSec = vestingPeriodSec;
+    vestingSchedule.vestingCliffStartTime = vestingStartTime.add(vestingCliff);
+    vestingSchedule.vestingStartTime = vestingStartTime;
+
+    beneficiary = vestingBeneficiary;
+    revocable = vestingRevocable;
+    revoker = vestingRevoker;
+    maxPausePeriod = vestingMaxPausePeriod;
+  }
+
+  /**
+   * @notice Returns if the vesting has been revoked or not.
+   */
+  function isRevoked() public view returns (bool) {
+    return revokeTime > 0;
+  }
+
+  /**
+   * @notice Transfers gold from the vesting back to beneficiary.
+   * @param amount the requested gold amount
+   */
+  function withdraw(uint256 amount) external nonReentrant onlyBeneficiary {
+    bool isPaused = block.timestamp < pauseEndTime;
+    require(!isPaused, "Withdrawals only allowed in the unpaused state");
+    require(amount > 0, "withdrawable amount must be greater than zero");
+
+    uint256 vestedAmount;
+    if (isRevoked()) {
+      vestedAmount = vestedBalanceAtRevoke;
+    } else {
+      vestedAmount = getVestedTotalBalance();
+    }
+
+    require(
+      vestedAmount.sub(totalWithdrawn) >= amount,
+      "Required amount is above withdrawable vested amount"
+    );
+    require(
+      getRemainingUnlockedBalance() >= amount,
+      "Insufficient unlocked balance to withdraw the amount from"
+    );
+    totalWithdrawn = totalWithdrawn.add(amount);
+    require(getGoldToken().transfer(beneficiary, amount), "Withdrawal of gold failed");
+    if (getRemainingTotalBalance() == 0) {
+      selfdestruct(beneficiary);
+    }
+  }
+
+  /**
+   * @notice Refund revoker and beneficiary after the vesting has been revoked.
+   */
+  function refundAndFinalize() external nonReentrant onlyRevokerAndRevoked {
+    require(getRemainingLockedBalance() == 0, "Some of the vested gold is still locked");
+    uint256 beneficiaryAmount = vestedBalanceAtRevoke.sub(totalWithdrawn);
+    require(
```

Comment:
check if `beneficiaryAmount` is bigger than 0 before doing a transfer

Reply:
ok, included